lib/asciidoctor/opal_ext.rb

```diff
@@ -1,64 +1,104 @@
+  var isNode = typeof process === 'object' && typeof process.versions === 'object' && process.browser != true,
+      isElectron = typeof navigator === 'object' && typeof navigator.userAgent === 'string' && typeof navigator.userAgent.indexOf('Electron') !== -1,
+      isBrowser = typeof window === 'object',
+      isNashorn = typeof Java === 'object' && Java.type,
+      isRhino = typeof java === 'object',
+      isPhantomJS = typeof window === 'object' && typeof window.phantom === 'object',
       isWebWorker = typeof importScripts === 'function',
-      isSpiderMonkey = typeof JSRuntime !== 'undefined',
-      platform = '',
-      engine = '',
-      framework = '',
-      ioModule = '';
+      isSpiderMonkey = typeof JSRuntime === 'object',
+      platform,
+      engine,
+      framework,
+      ioModule;
 
-  // Auto-detect the platform, engine and framework of the JavaScript environment
-  // NOTE: The order of the if statements is important because, for instance, 'module' will be defined in a Browserify environment
-  if (isBrowser) {
-    platform = 'browser';
-    if (isPhantomJS) {
-      framework = 'phantomjs';
-    }
+  // Load common modules
+  Opal.load("pathname");
+  Opal.load("base64");
+
+  // TODO: Use a global variable to configure Asciidoctor.js when running in a browser environment ?
+  if (typeof ajsConfig === 'object' && typeof ajsConfig.runtime === 'object') {
+    var runtime = ajsConfig.runtime;
+    platform = runtime.platform;
+    engine = runtime.engine;
+    framework = runtime.framework;
+    ioModule = runtime.ioModule;
   }
-  else if (isNode) {
-    platform = 'node';
-    engine = 'v8';
-    if (isElectron) {
-      framework = 'electron';
+
+  if (typeof platform === 'undefined') {
+    // Try to auto-detect the platform, engine and framework of the JavaScript runtime
+    if (isNode) {
+      platform = platform || 'node';
+      engine = engine || 'v8';
+      if (isElectron) {
+        framework = framework || 'electron';
+      }
+    }
+    else if (isNashorn) {
+      platform = plaform || 'java';
+      engine = engine || 'nashorn';
+    }
+    else if (isRhino) {
+      platform = platform || 'java';
+      engine = engine || 'rhino';
+    }
+    else if (isSpiderMonkey) {
+      platform = platform || 'standalone';
+      framework = framework || 'spidermonkey';
+    }
+    else if (isBrowser) {
+      platform = platform || 'browser';
+      if (isPhantomJS) {
+        framework = framework || 'phantomjs';
+      }
+    }
+    // NOTE: WebWorker are not limited to browser
+    if (isWebWorker) {
+      framework = framework || 'webworker';
     }
-    Opal.load("nodejs");
-    Opal.load("pathname");
-    Opal.load("base64");
   }
-  else if (isNashorn) {
-    platform = 'java';
-    engine = 'nashorn';
+
+  if (typeof platform === 'undefined') {
+    throw new Error('Unable to auto-detect the platform! Please configure Asciidoctor.js');
```

Comment:
If we can't determine the `platform` some features (mostly I/O features like include macro, embedded images, etc...) will not work.

@mojavelinux Do you think it's a good idea to throw an exception or should we just "ignore" this issue ? Initially I wanted to print a warning but since we do not know the "platform", we do not know how to print a log...